notebooks/_06-Architecture.md

```diff
@@ -0,0 +1,429 @@
+
+### 15.2 Is there a plan for sharing the software with others?
+- **File upload?**
+- **Upload to package repository?**
+
+All project related material including code, documentation, presentation, environment files, and datasets will be uploaded to the project github repository.
+
+### 15.3 Will the project have an associated website? If so is there a plan for its development?
+The only "website" for the project will be its github repository.
+
+### 15.4 Is there a plan for producing any written or visual materials that others may wish to view?
+- **Final or interim project reports**
+- **Tables of key project results**
+- **Graphs, diagrams, images from the project, e.g.:**
+ - **Plots of system performance**
+ - **Model checking plots**
+
+* All the plots will be generated using functions from the source code. Additional plots as needed will be generated
+using matplotlib/seaborn. 
+* Tables will be generated as markdown tables, html tables, or any other way if found easier
+* Final presentation could be developed either on powerpoint or as a set of slides using jupyter
+
+## 16 Description of maintenance plan:
+### 16.1 How will the system / software be maintained after initial creation?
+
+The source code of the project will be improved/expanded by adding more capabilities as deemed needed during grad
+school
+
+### 16.2 What needs to be done for maintenance?
+The source code will be modified following the github workflow.
+
+### 16.3 Who will perform needed maintenance activities?
+Hassan/Amine, and other interested students in Joan's group.
+
+## 17. Description of project tools:
+### 17.1 What are all the tools and major external systems upon which the project depends?
+Standard scientific packages such as: numpy, pandas, scipy, pylogit, causalgraphicalmodels(or others),
+seaborn/matplotlib, scikitlearn, pyro, fitter, pytest. This list is not exhaustive and could grow.
+
+### 17.2 Is there evidence that we understand the basics of how all of those tools work?
+- **Have we read any/all relevant documentation and instruction materials?**
+- **Have we taken any basic training courses in any of these tools?**
+
+Each team members is individually familiar working with the majority of these packages. Collectively, the team is
+familiar working with all of these packages. If additional packages are needed and no team member has the expertise
+needed to work with them, the team will make sure to read any available documentation or use examples available. 
+
+## 18. Description of project risks and mitigation efforts:
+### 18.1  There should be some evidence for, and of, project feasibility:
+- **How do we know that every part of the architecture will work?**
+- **Based on the architecture, what are all needed resources?**
+
+The architecture is mainly based on third party dependencies that have proven to work. The needed resources will be
```

Comment:
The first part of the question is "How do we know that every part of the architecture will work?"

Consider Q1. of the architecture doc. There, two main steps are listed for the project.

How do you know that each of these two steps will work? More specifically,
  - What precise papers support the belief that they'll work?
  - What prototype work (feel free to link pull requests / commits) supports the belief that the tasks will work?
  - What logical arguments support the belief that these tasks will work? 
     - If there is no evidence besides the logic, what is the smallest prototype that can be done to provide some evidence in support of the logic

Similar "how do you know it will work" questions apply to Q2 and Q3 of the architecture doc.

Reply:
I will be addressing this in detail after our meeting.

Reply:
@timothyb0912 I think this will need to be revisited based on discussions from last week and from our weekly mtg today